Project.toml

```diff
@@ -1,7 +1,7 @@
 name = "HydroPowerModels"
 uuid = "1bf2e10f-7293-4f36-bafb-f7584ca75eae"
 authors = ["andrewrosemberg <andrewrosemberg@gmail.com>"]
-version = "0.4.2"
+version = "0.5.2"
```

Comment:
Has this package been registered? If not, I think all of these versions changes are cosmetic. They don't actually do or enforce anything?

Reply:
Good question, the package is not registered. I update the version since @andrewrosemberg 's request, but I didn't think much about it.
Anyway, thanks for the correction, I wasn't resetting the last index when updating the intermediary one.

Reply:
Not yet registered! I got used to doing this just from internal packages in my work but it is also nice to keep track. 

I was planning on registering once we have an updated documentation, but I am cool either way if you guys want to register already.

Reply:
Btw since @odow is most familiar with the infinity horizon, once he approves, feel free to merge! Overall looks good to me documentation, API and basic tests wise.